### PR TITLE
234 investigate if an all in one docker image is viable

### LINKED ISSUE #234: Investigate if an all in one Docker image is viable

### PR DESCRIPTION
**Pull request by lpinon**
closes #234

### CODE REVIEW

File: setup/README.asciidoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- GCloud 
+- Kubectl
+- Terraform 
+- GitHub
+
+**Plus (+)** other tools like `unzip`, `git`, `wget` & `python`
+
+## How to use
+
+. **Build** the image `docker build -t hangar -f .\setup\Dockerfile .`
```
Comment (albertdb) [inline setup/README.asciidoc:25]:
```suggestion
. **Build** the image `docker build -t hangar -f ./setup/Dockerfile .`
```

Devonfw requires command to be Bash compatible.

File: setup/README.asciidoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+. **Execute** a command inside the image `docker run --rm -it hangar gcloud auth login` or open an **interactive shell** `docker run --rm -it hangar bash`
+
+IMPORTANT: The container is Stateless -- all the information (credentials, configurations, etc.) created inside the session will be lost at the end of the lifecycle of the container.
```
Comment (albertdb) [inline setup/README.asciidoc:28]:
```suggestion
WARNING: The container is stateless -- all the information (credentials, configurations, etc.) created inside the session will be lost at the end of the lifecycle of the container. 

```

File: setup/README.asciidoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+## How to use
+
+. **Build** the image `docker build -t hangar -f .\setup\Dockerfile .`
+. **Execute** a command inside the image `docker run --rm -it hangar gcloud auth login` or open an **interactive shell** `docker run --rm -it hangar bash`
+
+IMPORTANT: The container is Stateless -- all the information (credentials, configurations, etc.) created inside the session will be lost at the end of the lifecycle of the container. 
+
+TIP: If you want to have **persistence between executions** or **map the configuration files from your local host** check the next section for mapping volumes.
+
+## Volumes Configuration
```
Comment (albertdb) [inline setup/README.asciidoc:32]:
```suggestion
## Running container with persistence

```

**Review by albertdb — CHANGES_REQUESTED** [review]

File: setup/README.asciidoc [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# All-in-one container image for running Hangar.
```
Comment (albertdb) [inline setup/README.asciidoc:1]:
```suggestion
# All-in-one container image for running Hangar
```

File: setup/README.asciidoc [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+____
+As a Developer (Client, Capgemini, Third Party, anyone on the Web) I want to be able to download a single executable so I will be able to run it without an installation process (run the executable directly from the “Downloads” directory if necessary) and have all the CLIs for Azure, AWS & GCloud available.
+____
+
+
+This solution uses a *Dockerfile* to create an image with all the required tools ready to use. 
+
+
+## Included Tools
```
Comment (albertdb) [inline setup/README.asciidoc:11]:
```suggestion
## Included tools

```

File: setup/README.asciidoc [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Following toolkits are included in the solution:
```
Comment (albertdb) [inline setup/README.asciidoc:13]:
```suggestion
The following toolkits are included in the solution:
```

**Review by albertdb — APPROVED** [review]

File: setup/README.asciidoc [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+TIP: If you want to have **persistence between executions** or **map the configuration files from your local host** check the next section for mapping volumes.
+
+## Running container with persistence
+
+IMPORTANT: Before mounting the volumes make sure all the folders exist, Rancher will not create them by default. 
+
+### Windows
+
+```
+docker run --rm -it -v /Users/$env:UserName/AppData/Roaming/gcloud:/root/.config/gcloud -v /Users/$env:UserName/.aws:/root/.aws -v /Users/$env:UserName/.azure:/root/.azure -v /Users/$env:UserName/.kube:/root/.kube -v "/Users/$env:UserName/AppData/Roaming/GitHub CLI:/root/.config/gh" -v /Users/$env:UserName/.ssh:/root/.ssh hangar bash
```
Comment (ultymatom) [inline setup/README.asciidoc:41]: I think there is an error on the paths,
on powershell, it should be :
```suggestion
docker run --rm -it -v C:\Users\$env:UserName\AppData\Roaming\gcloud\:/root/.config/gcloud -v C:\Users\$env:UserName\.aws:/root/.aws -v C:\Users\$env:UserName\.azure:/root/.azure -v C:\Users\$env:UserName\.kube:/root/.kube -v "C:\Users\$env:UserName\AppData\Roaming\GitHub CLI:/root/.config/gh" -v C:\Users\$env:UserName\.ssh:/root/.ssh hangar bash

```

File: setup/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,54 @@
+FROM python:3.7-slim as base
+
+RUN apt-get update && apt-get install --no-install-recommends curl unzip git wget -y && apt-get clean && rm -rf /var/lib/apt/lists/*
```
Comment (ultymatom) [inline setup/Dockerfile:3]: I needed to install less to be able to make it work
```suggestion
RUN apt-get update && apt-get install --no-install-recommends curl unzip git wget less -y && apt-get clean && rm -rf /var/lib/apt/lists/*

```

File: setup/README.asciidoc (hunk 7 above)
Comment (lpinon) [reply setup/README.asciidoc:42]: Added as alternative

**Review by ultymatom — APPROVED** [review]

**Review by albertdb — APPROVED** [review]

**Pull request merged**